bitcoin/script/miniscript.cpp

```diff
@@ -170,27 +202,44 @@ Type ComputeType(NodeType nodetype, Type x, Type y, Type z, const std::vector<Ty
             (x & z & "e"_mst).If(x << "s"_mst || y << "f"_mst) | // e=e_x*e_z*(s_x+f_y)
             (x & y & z & "m"_mst).If(x << "e"_mst && (x | y | z) << "s"_mst) | // m=m_x*m_y*m_z*e_x*(s_x+s_y+s_z)
             (z & (x | y) & "s"_mst) | // s=s_z*(s_x+s_y)
-            "x"_mst; // x
-        case NodeType::MULTI: return "Bnudems"_mst;
+            "x"_mst | // x
+            ((x | y | z) & "ghij"_mst) | // g=g_x+g_y+g_z, h=h_x+h_y+h_z, i=i_x+i_y+i_z, j=j_x+j_y_j_z
+            "k"_mst.If(((x & y & z) << "k"_mst) &&
+                !(((x << "g"_mst) && (y << "h"_mst)) ||
+                ((x << "h"_mst) && (y << "g"_mst)) ||
+                ((x << "i"_mst) && (y << "j"_mst)) ||
+                ((x << "j"_mst) && (y << "i"_mst)))); // k=k_x*k_y*k_z* !(g_x*h_y + h_x*g_y + i_x*j_y + j_x*i_y)
+        case NodeType::MULTI: return "Bnudemsk"_mst;
         case NodeType::THRESH: {
             bool all_e = true;
             bool all_m = true;
             uint32_t args = 0;
             uint32_t num_s = 0;
+            Type acc_tl = "k"_mst;
             for (size_t i = 0; i < sub_types.size(); ++i) {
                 Type t = sub_types[i];
                 if (!(t << (i ? "Wdu"_mst : "Bdu"_mst))) return ""_mst; // Require Bdu, Wdu, Wdu, ...
                 if (!(t << "e"_mst)) all_e = false;
                 if (!(t << "m"_mst)) all_m = false;
                 if (t << "s"_mst) num_s += 1;
                 args += (t << "z"_mst) ? 0 : (t << "o"_mst) ? 1 : 2;
+                acc_tl = ((acc_tl | t) & "ghij"_mst) |
+                    // Thresh contains a combination of timelocks if it has threshold > 1 and
+                    // it contains two different children that have different types of timelocks
+                    // Note how if any of the children don't have "k", the parent also does not have "k"
+                    "k"_mst.If(((acc_tl & t) << "k"_mst) && ((k <= 1) ||
```

Comment:
This might be a bit tricky to review. Let me know if there is some better expression for this.